AGENTS.md

```diff
@@ -0,0 +1,260 @@
+- DO NOT modify any generated code or `.api` files manually
+- DO NOT expose internal modules or APIs to SDK users
+- DO NOT modify root `build.gradle` or module `build.gradle` files unless specifically asked
+- DO NOT commit configuration files with actual credentials
+- DO NOT include files with credentials in generated code
+- DO NOT use Android features unavailable in API 21+ unless compatibility checks are included
+- DO NOT bypass the DI system with direct instantiation of internal classes
+
+## Android-Specific Considerations
+- **Minimum SDK**: API 21 (Android 5.0)
+- **Target SDK**: API 33+ (follow current Android recommendations)
+- **Kotlin Version**: 1.8.21 - when adding new dependencies, ensure they're compatible with this Kotlin version
+- **Compose BOM**: 2023.03.00 (compatible with compileSdk 33)
+- Always check API level compatibility when using newer Android features
+- Use AndroidX libraries instead of deprecated support libraries
+- Handle Android lifecycle properly (Activities, Fragments, Services)
+- Consider memory implications of long-running operations
+- Use appropriate threading (Main thread for UI, background for network/disk)
+- Follow Android security best practices (no hardcoded secrets, proper permissions)
+- When adding new dependencies, verify Kotlin compatibility in `buildSrc/src/main/kotlin/io.customer/android/Versions.kt`
+
+## Memory and Performance
+- Use appropriate collection types (List vs Array, mutable vs immutable)
+- Avoid memory leaks by properly managing listeners and callbacks
+- Use weak references for callbacks that reference Activities/Contexts
+- Consider object pooling for frequently created objects
+- Profile memory usage during development
+- Use ProGuard/R8 rules appropriately for release builds
+- Monitor SDK binary size impact
+
+## Threading and Concurrency
+- Use Kotlin coroutines for asynchronous operations
+- Prefer structured concurrency over raw threads
+- Always specify appropriate Dispatchers (Main, IO, Default)
+- Handle cancellation properly in coroutines
+- Use thread-safe data structures when sharing state
+- Document threading requirements in public APIs
+- Test concurrent code thoroughly
+
+## Testing Strategy
+- Write both JVM unit tests and Android instrumentation tests
+- Use `UnitTest` base class for standard unit tests
+- Use `AndroidTest` base class for instrumentation tests
+- Use `RobolectricTest` for Android framework testing without device
+- Mock dependencies using MockK framework
+- Use test doubles and stubs from `common-test` module
+- Follow AAA pattern (Arrange, Act, Assert)
+- Name tests with `testMethodName_whenCondition_thenExpectedResult` format
+
+## Committing and Git Workflow
+- Use `lefthook` for git hooks (install with `lefthook install`)
+- Run `make format` before committing
+- Run `make lint` to ensure code quality
+- Follow conventional commit messages
+- Use feature branches off `develop` branch
+- Create pull requests for all changes
```

Comment:
Just want to double-check, this won't automatically start creating commits and PRs without consent, right? 😅